From 1333df0940a270cc2f6a1d509ac87029f8da7005 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 21 Oct 2026 08:25:31 -0700
Subject: [PATCH] [cmake] Avoid redundant dav1d rebuilds with depends

Reuse a matching dav1d from the depends prefix on Linux and FreeBSD.
Rebuilding it unnecessarily also forces an internal FFmpeg rebuild.
---
 cmake/modules/FindDav1d.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/FindDav1d.cmake b/cmake/modules/FindDav1d.cmake
index a1b623e..4cb7776 100644
--- a/cmake/modules/FindDav1d.cmake
+++ b/cmake/modules/FindDav1d.cmake
@@ -96,7 +96,8 @@ if(NOT TARGET LIBRARY::${CMAKE_FIND_PACKAGE_NAME})
   SEARCH_EXISTING_PACKAGES()
 
   if((${${CMAKE_FIND_PACKAGE_NAME}_SEARCH_NAME}_VERSION VERSION_LESS ${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER} AND ENABLE_INTERNAL_DAV1D) OR
-     (((CORE_SYSTEM_NAME STREQUAL linux AND NOT "webos" IN_LIST CORE_PLATFORM_NAME_LC) OR CORE_SYSTEM_NAME STREQUAL freebsd) AND ENABLE_INTERNAL_DAV1D))
+     ((CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd) AND
+      NOT KODI_DEPENDSBUILD AND ENABLE_INTERNAL_DAV1D))
     message(STATUS "Building ${${CMAKE_FIND_PACKAGE_NAME}_MODULE_LC}: \(version \"${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER}\"\)")
 
     cmake_language(EVAL CODE "
-- 
2.53.0

